test/unittests/test_parse_de.py

```diff
@@ -33,55 +37,218 @@ def tearDownModule():
-                         2)
+    def test_spaces(self):
+        self.assertEqual(normalize("  dies   ist  ein    test", lang="de-de"),
+                         "dies ist ein test")
+        self.assertEqual(normalize("  dies   ist  ein    test  ",
+                                   lang="de-de"), "dies ist ein test")
+
+    def test_numbers(self):
+        self.assertEqual(
+            normalize("dies ist eins zwei drei test", lang="de-de"),
+            "dies ist 1 2 3 test")
+        self.assertEqual(
+            normalize("es ist vier fünf sechs test", lang="de-de"),
+            "es ist 4 5 6 test")
+        self.assertEqual(
+            normalize("es ist sieben acht neun test", lang="de-de"),
+            "es ist 7 8 9 test")
+        self.assertEqual(
+            normalize("es ist sieben acht neun test", lang="de-de"),
+            "es ist 7 8 9 test")
+        self.assertEqual(
+            normalize("dies ist zehn elf zwölf test", lang="de-de"),
+            "dies ist 10 11 12 test")
         self.assertEqual(
-            extract_number("dies ist der dritte Test", lang="de-de"), 3)
+            normalize("dies ist dreizehn vierzehn test", lang="de-de"),
+            "dies ist 13 14 test")
         self.assertEqual(
-            extract_number("dies ist der Test Nummer 4", lang="de-de"), 4)
+            normalize("dies ist fünfzehn sechzehn siebzehn", lang="de-de"),
+            "dies ist 15 16 17")
+        self.assertEqual(
+            normalize("dies ist achtzehn neunzehn zwanzig", lang="de-de"),
+            "dies ist 18 19 20")
+
+
+class TestExtractNumber(unittest.TestCase):
+    
+    def test_extract_number(self):
+
+        self.assertEqual(extract_number("dies ist der 1. von 100 Tests"), 100)
+
+        self.assertEqual(extract_number("dies ist Test 1.5"), 1.5)
+        
+        self.assertEqual(extract_number("dies ist 2 Test"), 2)
+
+        self.assertEqual(extract_number("dies ist Test 3 von 4"), 3)
+
+        self.assertEqual(extract_number("dies ist der Test Nummer 4"), 4)
+
+        self.assertEqual(extract_number("drei Tassen"), 3)
+
+        self.assertEqual(extract_number("dreißig tausend vier hundert zwanzig"), 
+                                        30420)
+
+        self.assertEqual(extract_number("dreißig tausend vier hundert zwanzig komma zwei"),
+                                        30420.2)
+
+        self.assertEqual(extract_number("billionen jahre später"),
+                                        1000000000000)
+
+        self.assertEqual(extract_number("frisörtermin in einer stunde"), 1)
+
+        self.assertEqual(extract_number("frisörtermin in 1 stunde"), 1)
+
+        self.assertEqual(extract_number("minus dreißig grad"), -30)
+
+        self.assertEqual(extract_number("das macht null sinn"), 0)
+
+        self.assertEqual(extract_number("eins punkt drei vier"), 1.34)
+
+        self.assertEqual(extract_number("eins komma drei vier"), 1.34)
```

Comment:
Is this converted to a dotted notation, not 1,34?

Reply:
this is parsing, not formatting. And "1,34" is not a number to work with